src/backend_model_instance.h

```diff
@@ -125,6 +125,27 @@ class TritonModelInstance {
 
   MetricModelReporter* MetricReporter() const { return reporter_.get(); }
 
+  // Directly call from C API, so arguments are in the same style
+  void SetMemoryUsage(BufferAttributes** memory_usage, uint32_t usage_size)
```

Comment:
This could probably be implicitly converted to a `boost::span<BufferAttributes*>` if you wanted to not use the C style in implementation.